src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -482,9 +494,20 @@ public override async ValueTask<int> ReadAsync(Memory<byte> buffer, Cancellation
                             _reader = (JSObject)body.Invoke("getReader");
                         }
                     }
-                    catch (JSException)
+                    catch (OperationCanceledException oce)
+                    {
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            throw CancellationHelper.CreateOperationCanceledException(oce, cancellationToken);
+                        }
+                        throw;
+                    }
+                    catch (JSException jse)
                     {
-                        cancellationToken.ThrowIfCancellationRequested();
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            throw CancellationHelper.CreateOperationCanceledException(jse, cancellationToken);
+                        }
                         throw;
```

Comment:
We should also probably `throw new System.Net.Http.HttpRequestException(jse.Message);` here

Reply:
To address https://github.com/dotnet/runtime/issues/54160